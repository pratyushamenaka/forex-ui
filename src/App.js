import React, {Component} from 'react';
import logo from './logo.svg';
import moment from 'moment';
import './App.css';
import Background from './forex.png';


import Axios from 'axios';

class stockMarket extends Component {

 constructor(props) {
    super(props)
    this.state = {
      error: false,
      loading: true,
      items: null
    }
 }

 componentDidMount(){
    Axios.get('http://localhost:9090/api/calculate-profit', {
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Content-Type': 'application/json',
        },
        crossOrigin: true
      })
    .then(result => {
      this.setState({
        items: result.data,
        loading: false,
        error: false
      }) 
    })
    .catch(error => {
      this.setState({
         error: true
       });
    })
  }


  
  renderDetails(items) {
    return items.map(item => (
      <div>
        <ul className='stock-list'>
          <li>{moment(item.stockDate).format('DD-MMM-YY')}</li>
          <li>{item.currency}</li>
          <li>
            <span>Buy</span>
            <span>Sell</span>
          </li>
          <li>
            <span>${item.buyPrice}</span>
            <span>${item.sellPrice}</span>
          </li>
          <li>
            <span>{moment(item.buyTime, 'HHmm').format('hh:mma')}</span>
            <span>{moment(item.sellTime, 'HHmm').format('hh:mma')}</span>
          </li>
          <li>Profit: ${item.profit}</li>
        </ul>
      </div>
      
    ));
  }


  render() {
    const { loading, items, error } = this.state;
    return (
      <div><h1> Stock Market App </h1>
      <div/>
      <div>
        {loading && <div>Loading data...</div>}
        {items && <div>{this.renderDetails(items)}</div>}
        {error && <div>Error loading data...</div>}
      </div>
      </div>

    );
  }
}

export default stockMarket;
