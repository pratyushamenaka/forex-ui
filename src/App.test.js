import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import App from './App';
import moment from 'moment';

const appData = [
    {
        "currency": "BTC",
        "buyTime": "03:30 PM",
        "stockDate": "20180507",
        "buyPrice": 33.56,
        "sellTime": "1230",
        "sellPrice": 37.01,
        "profit": 3.45
    },
    {
        "currency": "ETC",
        "buyTime": "0900",
        "stockDate": "20180507",
        "buyPrice": 1.45,
        "sellTime": "1700",
        "sellPrice": 2.15,
        "profit": 0.7
    },
    {
        "currency": "LTC",
        "buyTime": "1700",
        "stockDate": "20180507",
        "buyPrice": 14.15,
        "sellTime": "1245",
        "sellPrice": 15.03,
        "profit": 0.88
    }
]
describe('stock app', () => {

	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<App />, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	it('should check snapshot', () => {
		const component = shallow(<App/>)
		expect(component).toMatchSnapshot()
	})

	it('should render data', () => {
		const component = shallow(<App/>)
		component.setState({items: appData})
		const stockList = component.find('.stock-list').length
	  	expect(stockList).toBe(3);
	})

	it('should throw error', () => {
		const component = shallow(<App/>)
		component.setState({error: true})
		const stockList = component.find('.stock-list').length
	  	expect(stockList).toBe(0);
	})

})
