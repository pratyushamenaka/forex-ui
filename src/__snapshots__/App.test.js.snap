// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stock app should check snapshot 1`] = `ShallowWrapper {}`;
